src/main/java/com/czertainly/core/service/impl/CertificateServiceImpl.java

```diff
@@ -800,6 +808,7 @@ public void validate(Certificate certificate) {
     @ExternalAuthorization(resource = Resource.CERTIFICATE, action = ResourceAction.DETAIL)
     public CertificateValidationResultDto getCertificateValidationResult(SecuredUUID uuid) throws NotFoundException {
         Certificate certificate = getCertificateEntityWithAssociations(uuid);
+        if (certificate.isArchived()) throw new ValidationException("Cannot validate archived certificate with UUID %s.".formatted(uuid));
```

Comment:
Exception should not be thrown, it should behave like with validation disabled below. For example, user navigating to detail page of archived certificate would get exception when refreshing validation status on `Validation` tab.

It already handled by method `CertificateUtil.isValidationEnabled`.